### PR TITLE
Extend bufferpool test

### PR DESCRIPTION
**Pull request by willhug**
Summary: Adds on to the bufferpool test to write random bytes into
the buffers before putting them back into the sync.Pool

### CODE REVIEW

File: internal/bufferpool/bufferpool_test.go [hunk 1]
```diff
@@ -36,10 +37,23 @@ func TestBuffers(t *testing.T) {
 				buf := Get()
 				assert.Zero(t, buf.Len(), "Expected truncated buffer")
 				assert.NotZero(t, buf.Cap(), "Expected non-zero capacity")
+
+				b := getRandBytes()
```
Comment (prashantv) [inline internal/bufferpool/bufferpool_test.go]: do we really need random data? maybe just write a constant to simplify this
```go
const dummyData = "dummy data"

buf.WriteString(dummyData)
assert.Equal(t, buf.Len(), len(dummyData), "...")
```

File: internal/bufferpool/bufferpool_test.go [hunk 2]
```diff
@@ -36,10 +37,23 @@ func TestBuffers(t *testing.T) {
...
 			}
 			wg.Done()
 		}()
 	}
 	wg.Wait()
 }
+
+func getRandBytes() []byte {
+	b := make([]byte, rand.Intn(5000))
+	rand.Read(b)
```
Comment (prashantv) [inline internal/bufferpool/bufferpool_test.go]: technically read can return an error that is being ignored here. might be cleaner to not use this at all

File: internal/bufferpool/bufferpool_test.go [hunk 3]
```diff
@@ -21,13 +21,14 @@
...
 
 import (
-	"math/rand"
 	"sync"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+const dummyData = "dummy data"
```
Comment (akshayjshah) [inline internal/bufferpool/bufferpool_test.go]: Please underscore-prefix globals.

File: internal/bufferpool/bufferpool_test.go [hunk 4]
```diff
@@ -27,6 +27,8 @@ import (
...
```
Comment (prashantv) [inline internal/bufferpool/bufferpool_test.go]: i'd move the const into the test since it's not used anywhere else

**Pull request merged**